test/integration/shoots/networking/shoot_networking_test.go

```diff
@@ -0,0 +1,175 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+	Overview
+		- Tests the communication between all nodes of the shoot
+
+	AfterSuite
+		- Cleanup Workload in Shoot
+
+	Test: Create a nginx daemonset and test if it is reachable from each node.
+	Expected Output
+		- nginx's are reachable from each node
+ **/
+
+package networking_test
+
+import (
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"path/filepath"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"text/template"
+	"time"
+
+	gardencorev1alpha1 "github.com/gardener/gardener/pkg/apis/core/v1alpha1"
+	. "github.com/gardener/gardener/test/integration/shoots"
+
+	"github.com/gardener/gardener/pkg/logger"
+	. "github.com/gardener/gardener/test/integration/framework"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	kubeconfig     = flag.String("kubecfg", "", "the path to the kubeconfig  of the garden cluster that will be used for integration tests")
+	shootName      = flag.String("shoot-name", "", "the name of the shoot we want to test")
+	shootNamespace = flag.String("shoot-namespace", "", "the namespace name that the shoot resides in")
+	logLevel       = flag.String("verbose", "", "verbosity level, when set, logging level will be DEBUG")
+)
+
+const (
+	NetworkTestTimeout    = 1800 * time.Second
+	InitializationTimeout = 600 * time.Second
+	FinalizationTimeout   = 1800 * time.Second
+	DumpStateTimeout      = 5 * time.Minute
+)
+
+const (
+	nginxTemplateName = "network-nginx-deamonset.yaml.tpl"
+)
+
+func validateFlags() {
+	if !StringSet(*shootName) {
+		Fail("You should specify a shootName to test against")
+	}
+
+	if !StringSet(*kubeconfig) {
+		Fail("you need to specify the correct path for the kubeconfig")
+	}
+
+	if !FileExists(*kubeconfig) {
+		Fail("kubeconfig path does not exist")
+	}
+}
+
+var _ = Describe("Shoot network testing", func() {
+	var (
+		shootGardenerTest      *ShootGardenerTest
+		gardenerTestOperations *GardenerTestOperation
+		testLogger             *logrus.Logger
+
+		resourcesDir = filepath.Join("..", "..", "resources")
+
+		name      = "net-test"
+		namespace = "default"
+	)
+
+	CBeforeSuite(func(ctx context.Context) {
+		// validate flags
+		validateFlags()
+		testLogger = logger.AddWriter(logger.NewLogger(*logLevel), GinkgoWriter)
```

Comment:
Maybe we should consider adding a util function `NewLogger(*logLevel)` into utils and replace all occurance of addWriter? Also lets maybe use another name instead of testLogger, for example (networkTestLogger). 

```suggestion
		networkTestLogger = NewLogger(*logLevel)

Reply:
yes I will rename it.
I wanted to refactor this whole stuff when doing this issue: https://github.com/gardener/gardener/issues/1565
There I want to consolidate most of these things.

Reply:
okay makes sense.